agent/engine/dockerstate/json.go

```diff
@@ -55,47 +55,50 @@ func (state *DockerTaskEngineState) UnmarshalJSON(data []byte) error {
 	if err != nil {
 		return err
 	}
-	// reset it by just creating a new one and swapping shortly.
-	// This also means we don't have to lock for the remainder of this function
-	// because we are the only ones with a reference to clean
-	clean := newDockerTaskEngineState()
 
+	// run precheck; shake out all the errors before resetting state.
+	precheckState := newDockerTaskEngineState()
 	for _, task := range saved.Tasks {
-		clean.AddTask(task)
-	}
-	// add image states
-	for _, imageState := range saved.ImageStates {
-		clean.AddImageState(imageState)
+		precheckState.AddTask(task)
 	}
 	for id, container := range saved.IdToContainer {
 		taskArn, ok := saved.IdToTask[id]
 		if !ok {
 			return errors.New("Could not unmarshal state; incomplete save. There was no task for docker id " + id)
 		}
-		task, ok := clean.TaskByArn(taskArn)
+		task, ok := precheckState.TaskByArn(taskArn)
 		if !ok {
 			return errors.New("Could not unmarshal state; incomplete save. There was no task for arn " + taskArn)
 		}
-
 		// The container.Container pointers *must* match the task's container
 		// pointers for things to operate correctly; update them here
```

Comment:
nit: "update them here" can be removed

Reply:
made this update.  Will wait for tests to complete before merging to dev.